### ISSUE TITLE
User unable to scan sync QR code due to faulty local network detection

### ISSUE THREAD

**Issue by ilmotta** [status-im/status-react#21257]
# Bug Report

## Problem 

Sometimes, an iOS user can't scan pairing QR codes because the button "Enable network access" doesn't work, even after multiple presses. Discussions started here https://github.com/status-im/status-mobile/issues/21199#issuecomment-2339350588

As usual with pairing, we should make sure the experience is smooth because the user may be blocked during the onboarding flow (more critical than from within Settings).

## Expected behavior

We propose a simple solution:

- Rename the button `Enable network access` to `Verify network access` because the user may have already allowed in iOS settings, but we can't detect the setting reliably. For more details see https://github.com/status-im/status-mobile/issues/21199#issuecomment-2339729020
- Pressing on the button `Verify network access` should disable it until the preflight check event is finished (success or failure).
- Identify and document in this issue or in the code why the pairing preflight check can sometimes fail.

**Happy path scenarios:**

1. The user never allowed local network access to the app. When they open `Scan sync code > Scan QR code` tab, they should be able to press on the button `Verify network access`, see the pop-up `"Status" Would like to Access the Camera` and finally scan the QR code on the other device. This is how it works today, just the button text would be different. Same behavior for the `Enter sync code` tab, which means both tabs will now display the highlighted section below when the pairing preflight check wasn't yet made (in iOS).

<img src="https://github.com/user-attachments/assets/95b323d4-f881-4baf-a7f6-5dcb4246d6f3" width="350" />

2. The user already allowed local network access in the app's iOS settings. When they open `Scan sync code > Scan QR code` tab, they should see the highlighted section (in red from the previous happy path scenario). Once the button `Verify network access` is pressed, the user should be able to scan the QR code without further interactions (no pop-up). Same behavior for the `Enter sync code` tab.

**Sad path scenarios:**

1. If pressing on the button `Verify network access` fails, we should display an error in a toast saying `Failed to verify local network access`. This should be doable because the native call to `local-pairing-preflight-outbound-check` can return an error.

## Actual behavior

The user has either allowed or disallowed local network access in iOS settings for the Status app, but pressing on the button `Enable network access` does nothing.

https://github.com/user-attachments/assets/e90fc2d4-590e-46ca-9ee4-54fd0e28d458

### Reproduction

Multiple people were able to see the problem in iOS. Chances of seeing the problem are greater in a fresh install.

1. From the onboarding start, `Sync or recover from profile`
2. Press `Scan Sync Code`.
3. Press the button `Enable network access` (nothing should happen).

The same bug can be reproduced from the profile `Settings > Syncing`.

### Additional Information

- Status version: `develop` (checked at revision babcd96fb625aa843d373ced160d94fd6e54bbd9) 
- Operating System: iOS

**Comment by ilmotta** [comment]
@xAlisher, this issue is an important bug to tackle because it prevents some users from pairing. Beyond identifying why the failure might occur, the issue also proposes a few simple UX changes (in terms of development effort).

Could you verify the proposal is sound, or would you like to propose something different?

Thanks 🙌🏼

**Comment by xAlisher** [comment]
Could you please point me to the Figma file that mentions the "Enable network access" feature? I can't seem to find it anywhere. Should it be included in these flows? 

https://www.figma.com/design/V6nlpAWIf2e1XU8RJ9yQPe/Syncing-for-Mobile?node-id=675-177939&t=b42syt4CCpJWUnSB-1

https://www.figma.com/design/o4qG1bnFyuyFOvHQVGgeFY/Onboarding-for-Mobile?node-id=4121-312366&t=FZB4EtKcAmYX6Tsu-1

I'm curious why it's not included in our design files.

**Comment by ilmotta** [comment]
> I'm curious why it's not included in our design files.

@xAlisher, I don't know the answer to that, but I found the original PR that introduced these changes https://github.com/status-im/status-mobile/pull/16150. The PR explains why the changes were made, but I didn't see a designer in the loop. Perhaps somebody was involved in Discord, but I won't be able to find that.

From the original issue:

> [...] We need to create some helper functions to check if required. **We need a design**, something like requesting camera permission before scanning the QR, to ensure smooth and successful pairing.
>
> -- https://github.com/status-im/status-mobile/issues/16135

**Comment by ilmotta** [comment]
> Could you please point me to the Figma file that mentions the "Enable network access" feature? I can't seem to find it anywhere. Should it be included in these flows?

I would say so @xAlisher, it would be good to have Figma updated. I tried my best to describe via text the solution, but a visual representation would help. And Figma should still be kept in sync with the product for sure 👍🏼

**Comment by xAlisher** [comment]
https://www.figma.com/design/o4qG1bnFyuyFOvHQVGgeFY/Onboarding-for-Mobile?node-id=14770-48952&t=BaX8Gh6xYxNPZm09-1

Requires UI review by @pedro-et

**Comment by ilmotta** [comment]
> https://www.figma.com/design/o4qG1bnFyuyFOvHQVGgeFY/Onboarding-for-Mobile?node-id=14770-48952&t=BaX8Gh6xYxNPZm09-1
> 
> Requires UI review by pedro-et

@xAlisher just checking, is the UI approved?

**Comment by xAlisher** [comment]
Yes it's approved by @pedro-et

**Comment by churik** [comment]
I don't think that we have a capacity for it in 2.32, given the desired release cut date (and it is not assigned yet)
Deferring the issue, please be free to move it bak if you think there is a room for it @ilmotta

**Comment by pavloburykh** [comment]
**UPDATE:** 

@VolodLytvynenko @mariia-skrypnyk and I have faced the same issue on IOS (release 2.32/nightly builds). I think we need to investigate the reason why camera (scanner) does not open in case if there are any error during enabling local network on IOS. Otherwise user just stucks on scan QR screen and cannot proceed even with fallback sync flow, unless he we try to go with pasting sync code (which is not an obvious option). 

I have described the issue [here](https://github.com/status-im/status-mobile/issues/21894) but closed as duplicate of current one. 

https://github.com/user-attachments/assets/f33ede78-7388-43eb-8885-1d8b8dad8b96

**Comment by ulisesmac** [comment]
Taking this as I'm planning to solve it along with:
- https://github.com/status-im/status-mobile/issues/21566

**Comment by pavloburykh** [comment]
The issue is still valid. @anastasiyaig and I have faced it during release 2.33 testing. cc @ulisesmac @ilmotta
